main.py

```diff
@@ -85,90 +90,30 @@
 pin13.direction = Direction.INPUT
 pin13.pull = Pull.DOWN
 
-#allianceColor=(255, 0, 0)
-allianceColor=(0, 0, 255)
-
-def blinkingCube():
-    for cube in range(15, 20, 1):
-        pixels[cube]=(255, 0, 0)
-    for cube in range(0, 15):
-        pixels[cube]=(145, 0, 255)
-    pixels.show()
-    time.sleep(0.3)
-    for cube in range(15):
-        pixels[cube]=(0, 0, 0)
-    pixels.show()
-    time.sleep(0.2)
-    
-#while True:
-   # blinkingCube()
-
-
-
-def blinkingCone():
-    for cube in range(15, 20, 1):
-        pixels[cube]=(255, 0, 0)
-    for cone in range(0, 15):
-        pixels[cone]=(255, 165, 0)
-    pixels.show()
-    time.sleep(0.3)
-    for cone in range(15):
-        pixels[cone]=(0, 0, 0)
-    pixels.show()
-    time.sleep(0.2)
-
-#while True:
-    #blinkingCone()
-
-
-def enable():
-    def setAll(r, g, b):
-        for a in range(20):
-            pixels[a]=(r, g, b)
+'''
+#############################################
+######  color display functions
+#############################################
+'''
+def enabled(current_mode):
+    for countdown in range(steps, 1, -1):
+        read_conecube_mode()
+        if read_current_mode() != current_mode:
+            return
+        for i in range((20-conecube_length)):
+            pixels[i]=(0, 255*countdown/steps, 0)
         pixels.show()
+        time.sleep(wait)
+    for countup in range(1, steps, 1):
+        read_conecube_mode()
+        if read_current_mode() != current_mode:
+            return
+        for i in range((20-conecube_length)):
+            pixels[i]=(0, 255*countup/steps, 0)
+        pixels.show()
+        time.sleep(wait)
 
-        for enable in range(20):
-            pixels[enable]=(255, 0, 0)
-
-        for t in range(256):
-            setAll(0, t, 0)
-        for t in range(255, -1, -1):
-            setAll(0, t, 0)
-#while True:
-    #enable()
-
-
-count=0 
-def enabled():
-    #while True:
-
-        print('start of loop 1')
-        for alliance in range(15, 20, 1):
-            pixels[alliance]=(255, 0, 0)
-
-        steps=30
-        wait=0.04
-        for countdown in range(steps, 1, -1):
-            for enabled in range(0, 15):
-                pixels[enabled]=(0, 255*countdown/steps, 0)
-            pixels.show()
-            time.sleep(wait)
-            print('brightness down')
-
-        for countup in range(1, steps, 1):
-            print('in loop 1')
-            for enabled in range(0, 15):
-                pixels[enabled]=(0, 255*countup/steps, 0)
-            pixels.show()
-            time.sleep(wait)
-            print('brightness up')
-
-#while True:
- #   enabled()
-
-
-def noCode():
-    #while True:
+def RobotNotBooted():
```

Comment:
nit: python function names should start with lower case letters and be snake_case